src/main/java/com/patchworkmc/event/initialization/RegisterEventRegistrars.java

```diff
@@ -1,56 +1,65 @@
 
-import java.util.Map;
+import java.util.Set;
 import java.util.function.Consumer;
 
+import org.objectweb.asm.Handle;
+import org.objectweb.asm.Label;
 import org.objectweb.asm.MethodVisitor;
 import org.objectweb.asm.Opcodes;
 import org.objectweb.asm.Type;
 
-public class RegisterEventRegistrars implements Consumer<MethodVisitor> {
-	private Iterable<Map.Entry<String, String>> staticEventRegistrars;
-	private Iterable<Map.Entry<String, String>> instanceEventRegistrars;
+import com.patchworkmc.Constants.EventBus;
+import com.patchworkmc.Constants.Lambdas;
 
-	public RegisterEventRegistrars(Iterable<Map.Entry<String, String>> staticEventRegistrars, Iterable<Map.Entry<String, String>> instanceEventRegistrars) {
-		this.staticEventRegistrars = staticEventRegistrars;
-		this.instanceEventRegistrars = instanceEventRegistrars;
+public class RegisterEventRegistrars implements Consumer<MethodVisitor> {
+	private static final String EVENT_REGISTRAR_REGISTRY = "net/minecraftforge/eventbus/api/EventRegistrarRegistry";
+	private static final String EVENT_REGISTRAR_REGISTRY_INSTANCE_SIGNATURE = "Lnet/minecraftforge/eventbus/api/EventRegistrarRegistry;";
+	private final Set<String> staticClasses;
+	private final Set<String> instanceClasses;
+
+	public RegisterEventRegistrars(Set<String> staticClasses, Set<String> instanceClasses) {
+		this.staticClasses = staticClasses;
+		this.instanceClasses = instanceClasses;
 	}
 
 	@Override
 	public void accept(MethodVisitor method) {
-		for (Map.Entry<String, String> entry : staticEventRegistrars) {
-			String shimName = entry.getKey();
-			String baseName = entry.getValue();
-
-			method.visitFieldInsn(Opcodes.GETSTATIC, "net/minecraftforge/eventbus/api/EventRegistrarRegistry", "INSTANCE", "Lnet/minecraftforge/eventbus/api/EventRegistrarRegistry;");
-
-			method.visitLdcInsn(Type.getObjectType(baseName));
-
-			method.visitTypeInsn(Opcodes.NEW, shimName);
-			method.visitInsn(Opcodes.DUP);
-
-			method.visitMethodInsn(Opcodes.INVOKESPECIAL, shimName, "<init>", "()V", false);
-			method.visitMethodInsn(Opcodes.INVOKEINTERFACE, "net/minecraftforge/eventbus/api/EventRegistrarRegistry", "registerStatic", "(Ljava/lang/Class;Ljava/util/function/Consumer;)V", true);
+		// Making the instance a local variable and pushing it to the stack creates slightly more readable decompiled code.
+		Label start = new Label();
+		method.visitLabel(start);
+		// We store the instance as a local variable for more readable decompiled code.
+		method.visitFieldInsn(Opcodes.GETSTATIC, EVENT_REGISTRAR_REGISTRY, "INSTANCE", EVENT_REGISTRAR_REGISTRY_INSTANCE_SIGNATURE);
+		method.visitVarInsn(Opcodes.ASTORE, 1);
+
+		for (String className : staticClasses) {
+			method.visitVarInsn(Opcodes.ALOAD, 1); // Push the instance to the stack (1)
+			method.visitLdcInsn(Type.getObjectType(className)); // Push the class to the stack (2)
+
+			method.visitInvokeDynamicInsn("accept", "()Ljava/util/function/Consumer;", Lambdas.METAFACTORY,
```

Comment:
It seems like the code for writing `invokedynamic` instructions that create `Consumer`s / `BiConsumer`s is duplicated in a few places, it would be nice to make utility methods instead.